safekeeper/src/bin/safekeeper.rs

```diff
@@ -220,71 +230,77 @@ fn start_safekeeper(conf: SafeKeeperConf) -> Result<()> {
     let timeline_collector = safekeeper::metrics::TimelineCollector::new();
     metrics::register_internal(Box::new(timeline_collector))?;
 
-    let mut threads = vec![];
     let (wal_backup_launcher_tx, wal_backup_launcher_rx) = mpsc::channel(100);
 
     // Load all timelines from disk to memory.
     GlobalTimelines::init(conf.clone(), wal_backup_launcher_tx)?;
 
-    let conf_ = conf.clone();
-    threads.push(
-        thread::Builder::new()
-            .name("http_endpoint_thread".into())
-            .spawn(|| {
-                let router = http::make_router(conf_);
-                endpoint::serve_thread_main(
-                    router,
-                    http_listener,
-                    std::future::pending(), // never shut down
-                )
-                .unwrap();
-            })?,
-    );
+    // Keep handles to main tasks to die if any of them disappears. Probably
+    // replace it with JoinSet once we update tokio.
```

Comment:
What does this comment mean? `safekeeper` is already using tokio 1.27.0 like rest of the workspace. The minimum version in our `Cargo.toml` is not the real minimum because we have 1.27.0 in `Cargo.lock`.

Very much agree on the joinset usage, however it will not allow you to do the similar `(String, Result<Result<_, _>, JoinError>)` with `futures::future::FutureExt::map`... Perhaps you were referring to the `JoinSet::build_task(&mut self)` and `Builder::name` being unstable?

Alternative to this would be to have just plain futures here locally (each as they are `tokio::spawn(...).map(...)`) and then `tokio::try_join!(...)` over all of them. The `FuturesUnordered` provides very little.

Reply:
> The minimum version in our Cargo.toml is not the real minimum because we have 1.27.0 in Cargo.lock.

Oh, that was confusing, thanks. No, I meant specifically JoinSet.

So I think FuturesUnordered is ok here, removed comment.